pkg/cmd/server/middleware.go

```diff
@@ -11,21 +11,21 @@ import (
 // MiddlewareChain describes an ordered sequence of middlewares that can be modified
 // with one or more MiddlewareModification. This struct is used to facilitate the
 // creation and modification of gRPC middleware chains
-type MiddlewareChain struct {
-	chain []ReferenceableMiddleware
+type MiddlewareChain[T grpc.UnaryServerInterceptor | grpc.StreamServerInterceptor] struct {
```

Comment:
I think you can define an interface that implements this constraint and just reference it

Reply:
I like this is explicit. In which ways do you think adding a synthetic interface will improve the code?

Reply:
Just remove repeating it everywhere

Reply:
done in https://github.com/authzed/spicedb/pull/1391/commits/67c0fcb33a5165ad8b51b2c0400d63d7708d5db1